apis/manifests/v1alpha1/packagemanifest_types.go

```diff
@@ -2,31 +2,109 @@ package v1alpha1
 
 import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
 
-	pkoapisv1alpha1 "package-operator.run/apis/core/v1alpha1"
+	corev1alpha1 "package-operator.run/apis/core/v1alpha1"
+)
+
+const (
+	// Package Phase annotation to assign objects to a phase.
+	PackagePhaseAnnotation = "package-operator.run/phase"
+)
+
+const (
+	// PackageLabel contains the name of the Package from the PackageManifest.
+	PackageLabel = "package-operator.run/package"
+	// PackageInstanceLabel contains the name of the Package instance.
+	PackageInstanceLabel = "package-operator.run/instance"
 )
 
 // +kubebuilder:object:root=true
 type PackageManifest struct {
-	metav1.TypeMeta `json:",inline" yaml:",inline"`
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec PackageManifestSpec `json:"spec,omitempty" yaml:"spec,omitempty"`
+	Spec PackageManifestSpec `json:"spec,omitempty"`
 }
 
-// PackageManifestSpec represents the spec of the packagemanifest containing the details about phases and availability probes
+// PackageManifestScope declares the available scopes to install this package in.
+type PackageManifestScope string
+
+const (
+	// Cluster scope allows the package to be installed for the whole cluster.
+	// The package needs to default installation namespaces and create them.
+	PackageManifestScopeCluster PackageManifestScope = "Cluster"
+	// Namespace scope allows the package to be installed for specific namespaces.
+	PackageManifestScopeNamespaced PackageManifestScope = "Namespaced"
+)
+
+// PackageManifestSpec represents the spec of the packagemanifest containing the details about phases and availability probes.
 type PackageManifestSpec struct {
+	// Scopes declare the available installation scopes for the package.
+	// Either Cluster, Namespaced, or both.
+	Scopes []PackageManifestScope `json:"scopes"`
 	// Phases correspond to the references to the phases which are going to be the part of the ObjectDeployment/ClusterObjectDeployment.
-	Phases []PackageManifestPhase `json:"phases" yaml:"phases"`
+	Phases []PackageManifestPhase `json:"phases"`
 	// Availability Probes check objects that are part of the package.
 	// All probes need to succeed for a package to be considered Available.
 	// Failing probes will prevent the reconciliation of objects in later phases.
-	AvailabilityProbes []pkoapisv1alpha1.ObjectSetProbe `json:"availabilityProbes" yaml:"availabilityProbes"`
+	AvailabilityProbes []corev1alpha1.ObjectSetProbe `json:"availabilityProbes"`
 }
 
 type PackageManifestPhase struct {
 	// Name of the reconcile phase. Must be unique within a PackageManifest
-	Name  string `json:"name" yaml:"name"`
-	Class string `json:"class,omitempty" yaml:"class,omitempty"`
+	Name string `json:"name"`
+	// If non empty, phase reconciliation is delegated to another controller.
+	// If set to the string "default" the built-in controller reconciling the object.
```

Comment:
```suggestion
	// If set to the string "default" the remote-phase-manager will reconcile the object in the same cluster.
	// If set to the string "hosted-cluster" the remote-phase-manager will reconcile the object in the given target cluster.
```

Reply:
Nope, it's not part of the remote-phase-manager, it's embedded into the package-operator-manager:
https://github.com/package-operator/package-operator/blob/main/cmd/package-operator-manager/main.go#L177-L192